gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -1584,11 +1593,61 @@ public static void CREATE_OR_DROP_RESERVOIR_REGION(String reservoirRegionName,
     }
   }
 
+  /**
+   * Create or drop reservoir region for sampler.
+   *
+   * @param reservoirRegionName name of the reservoir region
+   * @param resolvedBaseName base table name with schema
+   * @param isDrop flag to indicate that the stored procedure is being invoked to drop the
+   *               reservoir region
+   * @param removeSampler flag to indicate that the cached sampler should also be removed
+   *                      along with the region. This flag will be a ignored if the `isDrop`
+   *                      flag is false.
+   */
+  public static void CREATE_OR_DROP_RESERVOIR_REGION_V2(String reservoirRegionName,
```

Comment:
Instead of `CREATE_OR_DROP_RESERVOIR_REGION_V2`, we considered keeping the stored procedure name `DROP_RESERVOIR_REGION_AND_REMOVE_SAMPLER` and keep the behavior of this new procedure accordingly. We dropped this approach as by that approach conflation may not work.

Note that the conflation won't work with this approach also for the regions created on older versions of the product.

@sumwale, @kneeraj, @ahshahid please share your thoughts on this.

Reply:
Right, that is where one of the other two approaches can be used. However, I think we can live with lack of conflation for this case which should not be anything more than some wasted effort.